src/axom/primal/operators/in_polygon.hpp

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ * \file in_polygon.hpp
+ *
+ * \brief Consists of methods that test whether a given query point is
+ * inside a given polygon.
+ *
+ * Uses a ray casting algorithm
+ */
+
+#ifndef AXOM_PRIMAL_IN_POLYGON_HPP_
+#define AXOM_PRIMAL_IN_POLYGON_HPP_
+
+// Axom includes
+#include "axom/config.hpp"
+
+#include "axom/primal/geometry/Point.hpp"
+#include "axom/primal/geometry/Polygon.hpp"
+
+// C++ includes
+#include <cmath>
+
+namespace axom
+{
+namespace primal
+{
+/*!
+ * \brief Computes the winding number for a point and a polygon
+ *
+ * \param [in] R The query point to test
+ * \param [in] P The Polygon object to test for containment
+ * \param [in[ strict If true, check for strict inclusion
+ * \param [in] EPS The tolerance level for collinearity
+ * 
+ * Uses an adapted ray-casting approach that counts quarter-rotation
+ * of vertices around the query point. 
+ *
+ * Directly uses algorithm in 
+ * Kai Hormann, Alexander Agathos, "The point in polygon problem for arbitrary polygons"
+ * Computational Geometry, Volume 20, Issue 3, 2001,
+ * 
+ * \return The integer winding number
+ */
+template <typename T>
+int winding_number(const Point<T, 2>& R,
+                   const Polygon<T, 2>& P,
+                   const bool strict = false,
+                   const double EPS = 1e-8)
+{
+  const int nverts = P.numVertices();
+
+  // If the query is a vertex, return a value interpreted
+  //  as "inside" by evenodd or nonzero protocols
+  if(R == P[0]) return !strict;
+
+  int winding_num = 0;
+  for(int i = 0; i < nverts; i++)
+  {
+    int j = (i == nverts - 1) ? 0 : i + 1;
+
+    if(P[j][1] == R[1])
```

Comment:
Should these equality tests be converted to fuzzy equality tests?
E.g.
```diff
-    if(P[j][1] == R[1])
+    if(axom::utilities::isNearlyEqual(P[j][1],R[1],EPS))
```
Or perhaps, the equality test is related to the `strict` parameter?

(throughout this function)

Reply:
It realistically should be a fuzzy equality, yes. The initial concern was that `EPS` in this context has a different interpretation than the `EPS` used for the determinant below. For these equalities, it's the literal distance between the points, but for the determinant, it's less direct, as it isn't the distance from the point to the edge.

For now though, I think the `isNearlyEqual` works.